docs/competing_risks.md

```diff
@@ -0,0 +1,278 @@
+
+It is often of interest to obtain the Cumulative Incidence Function (CIF) defined as :math:`Pr(T_i\leq t,D_i=d|X_{i})`.  The CIF is obtained using the forecasts from `LGBSurvivalModeler` and `LGBExitModeler` using the relationship
+
+.. math::
+	Pr(T_i\leq t,D_i=d|X_{i})&=\sum_{l=T_M+1}^{t}Pr(D_i=d|T_i=l,X_{i})Pr(T_i= l|X_{i})\textrm{, where}\\
+	Pr(T_i=l|X_{i})&=Pr(T_i\geq l-1|X_{i})-Pr(T_i\geq l|X_{i}).
+
+Note, :math:`Pr(T_i\geq 0 |X_{i})=1`.
+
+.. code-block:: python
+
+	# Obtain Pr(T=l)
+	tmp = concat([DataFrame(ones(survival_modeler_forecasts.shape[0]),index = survival_modeler_forecasts.index),survival_modeler_forecasts.iloc[:,:-1]], axis = 1)
+	Pr_T_eq_l_forecasts = tmp.values - survival_modeler_forecasts
+	
+	# Obtain the cumulative incidence function 
+	cumulative_incidence_forecasts = exit_modeler_forecasts.copy()
+	cumulative_incidence_forecasts = cumulative_incidence_forecasts * DataFrame(repeat(Pr_T_eq_l_forecasts.values, repeats = mdata['outcome'].nunique(), axis = 0)).values
+	cumulative_incidence_forecasts = cumulative_incidence_forecasts.cumsum(axis = 1)
+	cumulative_incidence_forecasts
+	```
+	                                                         Jul 2020      Aug 2020  ...      Jul 2040      Aug 2040
+	country-leader            Future outcome                                         ...                            
+	Afghanistan: Ashraf Ghani Same government type       1.197189e-02  2.748160e-02  ...  8.427979e-01  8.427992e-01
+	                          Change of government type  1.980397e-03  6.588223e-03  ...  1.549343e-01  1.549348e-01
+	                          Dissolution                3.909978e-06  1.647998e-05  ...  1.994732e-03  1.994732e-03
+	                          No exit                    1.402320e-17  3.637464e-17  ...  9.295344e-16  9.295362e-16
+	Albania: Rama             Same government type       7.641719e-03  1.520875e-02  ...  9.458562e-01  9.458959e-01
+	                                                          ...           ...  ...           ...           ...
+	Zambia: Lungu             No exit                    3.984779e-18  7.635237e-18  ...  6.894410e-16  6.897318e-16
+	Zimbabwe: Mnangagwa       Same government type       6.286339e-03  1.192818e-02  ...  7.680164e-01  7.682909e-01
+	                          Change of government type  5.620235e-04  1.644481e-03  ...  1.741905e-01  1.742979e-01
+	                          Dissolution                2.044422e-06  6.350160e-06  ...  4.357975e-04  4.357975e-04
+	                          No exit                    7.332364e-18  1.498897e-17  ...  1.108881e-15  1.109270e-15
+	
+	[776 rows x 242 columns]
+	```
+	
+	# Plot cumulative incidence function
+	plotdf = cumulative_incidence_forecasts.loc["USA: Trump",:].transpose()
+	plt.figure()
+	plt.ylim(bottom=0,top=1)
+	plotdf.plot()
+
+##### Cause-specific hazard probability
+
+You can also obtain the cause-specific probability defined as :math:`Pr(T_i=t,D_i=d|T_i\geq t,X_{i})`. The relationship is
```

Comment:
Please give an intuitive description of the cause-specific hazard probability, such as "This is the probability of a specific type of exit in a specific future time period, conditional on survival to that time period."

Reply:
Done